json/src/main/scala/com/gu/contentapi/json/CirceEncoders.scala

```diff
@@ -0,0 +1,113 @@
+
+  implicit def officeEncoder[A <: Office]: Encoder[A] = Encoder[String].contramap(o => o.name.toUpperCase)
+
+  /**
+    * We need special encoders for things with Maps, probably because of implicit divergence.
+    * TODO - surely there's a way to avoid doing this?!
+    */
+  implicit val blockMapEncoder = Encoder.instance[scala.collection.Map[String,Seq[Block]]] { m =>
+    val fields = m.toList.map {
+      case (k, v) => k -> Json.fromValues(v.map(_.asJson))
+    }
+    Json.fromFields(fields)
+  }
+
+  implicit val separatorLocationsEncoder = Encoder.instance[scala.collection.Map[String,Seq[Int]]] { s =>
+    val fields = s.toList.map {
+      case (k, v) => k -> Json.fromValues(v.map(_.asJson))
+    }
+    Json.fromFields(fields)
+  }
+
+  implicit val networkFrontEncoder = Encoder[NetworkFront]
+  implicit val dateTimeEncoder = genDateTimeEncoder
+  implicit val contentFieldsEncoder = Encoder[ContentFields]
+  implicit val editionEncoder = Encoder[Edition]
+  implicit val sponsorshipEncoder = Encoder[Sponsorship]
+  implicit val tagEncoder = Encoder[Tag]
+  implicit val assetEncoder = Encoder[Asset]
+  implicit val elementEncoder = Encoder[Element]
+  implicit val referenceEncoder = Encoder[Reference]
+  implicit val blockEncoder = Encoder[Block]
+  implicit val blocksEncoder = Encoder[Blocks]
+  implicit val rightsEncoder = Encoder[Rights]
+  implicit val crosswordEntryEncoder = Encoder[CrosswordEntry]
+  implicit val crosswordEncoder = Encoder[Crossword]
+  implicit val contentStatsEncoder = Encoder[ContentStats]
+  implicit val sectionEncoder = Encoder[Section]
+  implicit val debugEncoder = Encoder[Debug]
+  implicit val atomEncoder = AtomEncoder.genAtomEncoder
+  implicit val atomsEncoder = Encoder[Atoms]
+  implicit val contentEncoder = Encoder[Content]
+  implicit val mostViewedVideoEncoder = Encoder[MostViewedVideo]
+  implicit val packageEncoder = Encoder[Package]
+  implicit val itemResponseEncoder = Encoder[ItemResponse]
+  implicit val searchResponseEncoder = Encoder[SearchResponse]
+  implicit val editionsResponseEncoder = Encoder[EditionsResponse]
+  implicit val tagsResponseEncoder = Encoder[TagsResponse]
+  implicit val sectionsResponseEncoder = Encoder[SectionsResponse]
+  implicit val atomsResponseEncoder = Encoder[AtomsResponse]
+  implicit val packagesResponseEncoder = Encoder[PackagesResponse]
+  implicit val errorResponseEncoder = Encoder[ErrorResponse]
+  implicit val videoStatsResponseEncoder = Encoder[VideoStatsResponse]
+  implicit val atomsUsageResponseEncoder = Encoder[AtomUsageResponse]
+  implicit val removedContentResponseEncoder = Encoder[RemovedContentResponse]
+
+  def genDateTimeEncoder: Encoder[CapiDateTime] = Encoder.instance[CapiDateTime] { capiDateTime =>
+    Json.fromString(capiDateTime.iso8601)
+  }
+
+  /**
+    * TODO - I *will* write a pair of macros for encoding/decoding thrift union types, then delete all this stuff.
```

Comment:
You might want to talk to @emma-p and I if you do try and do this as we recently did https://github.com/guardian/marley/pull/3/files which was a bit hairy in places.